plugins/inputs/vsphere/README.MD

```diff
@@ -0,0 +1,350 @@
+  # vm_instances = true ## true by default
+
+  ############### Hosts ###############
+  ## Typical host metrics (if omitted or empty, all metrics are collected)
+  host_metric_include = [
+    "cpu.coreUtilization.average",
+    "cpu.costop.summation",
+    "cpu.demand.average",
+    "cpu.idle.summation",
+    "cpu.latency.average",
+    "cpu.readiness.average",
+    "cpu.ready.summation",
+    "cpu.swapwait.summation",
+    "cpu.usage.average",
+    "cpu.usagemhz.average",
+    "cpu.used.summation",
+    "cpu.utilization.average",
+    "cpu.wait.summation",
+    "disk.deviceReadLatency.average",
+    "disk.deviceWriteLatency.average",
+    "disk.kernelReadLatency.average",
+    "disk.kernelWriteLatency.average",
+    "disk.numberReadAveraged.average",
+    "disk.numberWriteAveraged.average",
+    "disk.read.average",
+    "disk.totalReadLatency.average",
+    "disk.totalWriteLatency.average",
+    "disk.write.average",
+    "mem.active.average",
+    "mem.latency.average",
+    "mem.state.latest",
+    "mem.swapin.average",
+    "mem.swapinRate.average",
+    "mem.swapout.average",
+    "mem.swapoutRate.average",
+    "mem.totalCapacity.average",
+    "mem.usage.average",
+    "mem.vmmemctl.average",
+    "net.bytesRx.average",
+    "net.bytesTx.average",
+    "net.droppedRx.summation",
+    "net.droppedTx.summation",
+    "net.errorsRx.summation",
+    "net.errorsTx.summation",
+    "net.usage.average",
+    "power.power.average",
+    "storageAdapter.numberReadAveraged.average",
+    "storageAdapter.numberWriteAveraged.average",
+    "storageAdapter.read.average",
+    "storageAdapter.write.average",
+    "sys.uptime.latest",
+  ]
+  # host_metric_exclude = [] ## Nothing excluded by default
+  # host_instances = true ## true by default
+
+  ############### Clusters ###############
+  # cluster_metric_include = [] ## if omitted or empty, all metrics are collected
+  # cluster_metric_exclude = [] ## Nothing excluded by default
+  # cluster_instances = true ## true by default
+
+  ############### Datastores ###############
+  # datastore_metric_include = [] ## if omitted or empty, all metrics are collected
+  # datastore_metric_exclude = [] ## Nothing excluded by default
+  # datastore_instances = false ## false by default for Datastores only
+
+
+  ############### Plugin Settings  ###############
+  ## separator character to use for measurement and field names (default: "_")
+  # separator = "_"
+
+  ## number of objects to retreive per query for realtime resources (vms and hosts)
+  ## set to 64 for vCenter 5.5 and 6.0 (default: 256)
+  # objects_per_query = 256
+
+  ## number of metrics to retreive per query for non-realtime resources (clusters and datastores)
+  ## set to 64 for vCenter 5.5 and 6.0 (default: 256)
+  # metrics_per_query = 256
+
+  ## number of go routines to use for collection and discovery of objects and metrics
+  # collect_concurrency = 1
+  # discover_concurrency = 1
```

Comment:
The resource that is scarce is connections, so I propose we rename this `max_discover_connections`.

BTW, did you see they added an option for this to the transport in Go 1.11 (MaxConnsPerHost)?  Should be pretty helpful, but we still target Go 1.9 so we can't use it yet.